bokeh/util/session_id.py

```diff
@@ -39,79 +42,167 @@
 
 __all__ = (
     'check_session_id_signature',
+    'check_token_signature',
     'generate_secret_key',
+    'generate_jwt_token',
     'generate_session_id',
+    'get_session_id',
+    'get_token_payload',
 )
 
 #-----------------------------------------------------------------------------
 # General API
 #-----------------------------------------------------------------------------
 
 def generate_secret_key() -> str:
-    """
-    Generate a new securely-generated secret key appropriate
-    for SHA-256 HMAC signatures. This key could be used to
-    sign Bokeh server session IDs for example.
-    """
+    ''' Generate a new securely-generated secret key appropriate for SHA-256
+    HMAC signatures.
+
+    This key could be used to sign Bokeh server session IDs, for example.
+    '''
     return _get_random_string()
 
 def generate_session_id(secret_key: Optional[bytes] = settings.secret_key_bytes(),
                         signed: bool = settings.sign_sessions()) -> str:
-    """Generate a random session ID.
-    Typically, each browser tab connected to a Bokeh application
-    has its own session ID.  In production deployments of a Bokeh
-    app, session IDs should be random and unguessable - otherwise
-    users of the app could interfere with one another.
-
-    If session IDs are signed with a secret key, the server can
-    verify that the generator of the session ID was "authorized"
-    (the generator had to know the secret key). This can be used
-    to have a separate process, such as another web application,
-    which generates new sessions on a Bokeh server. This other
-    process may require users to log in before redirecting them to
-    the Bokeh server with a valid session ID, for example.
+    ''' Generate a random session ID.
+
+    Typically, each browser tab connected to a Bokeh application has its own
+    session ID.  In production deployments of a Bokeh app, session IDs should be
+    random and unguessable - otherwise users of the app could interfere with one
+    another.
+    '''
+    session_id = _get_random_string()
+    if signed:
+        session_id = '.'.join([session_id, _signature(session_id, secret_key)])
+    return session_id
+
+def generate_jwt_token(session_id: str,
+                       secret_key: Optional[bytes] = settings.secret_key_bytes(),
+                       signed: bool = settings.sign_sessions(),
+                       extra_payload: Optional[Dict[str, Any]] = None,
+                       expiration: int = 300) -> str:
```

Comment:
Do we want to make this configurable?

Reply:
I think we should

Reply:
Oh and what default did we want, 300 seconds was completely arbitrary.

Reply:
5 minutes seems good to me!